odoo/addons/base/tests/test_res_users.py

```diff
@@ -118,6 +118,97 @@ def test_change_user_company(self):
+
+                Timesheets / Administrator
+                └── Timesheets / User: all timesheets
+                    └── Timehseets / User: own timesheets only
+
+                Project / Administrator
+                ├── Project / User
+                └── Timesheets / User: all timesheets
+
+                Field Service / Administrator
+                ├── Sales / Administrator
+                ├── Project / Administrator
+                └── Field Service / User
+        """
+        super().setUp()
+        ResGroups = self.env['res.groups']
+        cat_sales = self.env['ir.module.category'].create({'name': 'Sales'})
+        cat_project = self.env['ir.module.category'].create({'name': 'Project'})
+        cat_field_service = self.env['ir.module.category'].create({'name': 'Field Service'})
+        cat_timesheets = self.env['ir.module.category'].create({'name': 'Timesheets'})
+
+        # Sales
+        self.group_sales_user, self.group_sales_administrator = ResGroups.create([
+            {'name': name, 'category_id': cat_sales.id}
+            for name in ('User: All Documents', 'Administrator')
+        ])
+        self.sales_cat_field = name_selection_groups((self.group_sales_user | self.group_sales_administrator).ids)
+        self.group_sales_administrator.implied_ids = self.group_sales_user
+
+        # Timesheets
+        self.group_timesheets_user_own_timesheet, self.group_timesheets_user_all_timesheet, self.group_timesheets_administrator = ResGroups.create([
+            {'name': name, 'category_id': cat_timesheets.id}
+            for name in ('User: own timesheets only', 'User: all timesheets', 'Administrator')
+        ])
+        self.timesheet_cat_field = name_selection_groups((self.group_timesheets_user_own_timesheet |
+                                                          self.group_timesheets_user_all_timesheet |
+                                                          self.group_timesheets_administrator).ids
+                                                         )
+        self.group_timesheets_administrator.implied_ids += self.group_timesheets_user_all_timesheet
+        self.group_timesheets_user_all_timesheet.implied_ids += self.group_timesheets_user_own_timesheet
+
+        # Project
+        self.group_project_user, self.group_project_admnistrator = ResGroups.create([
+            {'name': name, 'category_id': cat_project.id}
+            for name in ('User', 'Administrator')
+        ])
+        self.project_cat_field = name_selection_groups((self.group_project_user | self.group_project_admnistrator).ids)
+        self.group_project_admnistrator.implied_ids = (self.group_project_user | self.group_timesheets_user_all_timesheet)
+
+        # Field Service
+        self.group_field_service_user, self.group_field_service_administrator = ResGroups.create([
+            {'name': name, 'category_id': cat_field_service.id}
+            for name in ('User', 'Administrator')
```

Comment:
(same above, no need to loop for 2 elements)
```suggestion
            {'name': 'User', 'category_id': cat_field_service.id},
            {'name': 'Administrator', 'category_id': cat_field_service.id},
```